### DISCUSSION TITLE
ERROR: Unhandled python exception (solved)

### DISCUSSION THREAD

**Discussion by bad-friday** [mesonbuild/meson#11374] [General]
I'm coming from Raspberry Pi runing Debian Bullseye.
I try to build libcamera following the instructions here: https://www.raspberrypi.com/documentation/computers/camera_software.html#building-libcamera-and-libcamera-apps
[/home/swe]meson build --buildtype=release -Dpipelines=raspberrypi -Dipas=raspberrypi -Dv4l2=true -Dgstreamer=disabled -Dtest=false -Dlc-compliance=disabled -Dcam=disabled -Dqcam=enabled -Ddocumentation=disabled 2>/tmp/meson.log
The Meson build system
Version: 1.0.0
Source dir: /home/swe/libcamera
Build dir: /home/swe/libcamera/build
Build type: native build
Project name: libcamera
Project version: 0.0.3
C compiler for the host machine: cc (gcc 10.2.1 "cc (Raspbian 10.2.1-6+rpi1) 10.2.1 20210110")
C linker for the host machine: cc ld.bfd 2.35.2
C++ compiler for the host machine: c++ (gcc 10.2.1 "c++ (Raspbian 10.2.1-6+rpi1) 10.2.1 20210110")
C++ linker for the host machine: c++ ld.bfd 2.35.2
Host machine cpu family: arm
Host machine cpu: armv7l
meson.build:33: WARNING: The sources disagree about the version: 0.0.0 != 0.0.3
Header "unistd.h" has symbol "issetugid" : NO 
Header "locale.h" has symbol "locale_t" : YES 
Header "stdlib.h" has symbol "secure_getenv" : YES 
Compiler for C supports arguments -Wno-c99-designator: NO 
Found pkg-config: /usr/bin/pkg-config (0.29.2)

meson.build:163:0: ERROR: Unhandled python exception

    This is a Meson bug and should be reported!

**Comment by bad-friday** [comment]
Here is the complete log including stderr
[Meson error.txt](https://github.com/mesonbuild/meson/files/10706694/Meson.error.txt)

**Reply by dcbaker** [reply]
Thanks! This is a relatively simple issue to fix on our end, but it's kinda surprising that you hit this, it's happening because we're finding a cmake binary, but it doesn't have any of the expected version outputs. Out of curiosity, what's the output of `cmake --version` on your system?

**Reply by hananbeer** [reply]
> Thanks! This is a relatively simple issue to fix on our end, but it's kinda surprising that you hit this, it's happening because we're finding a cmake binary, but it doesn't have any of the expected version outputs. Out of curiosity, what's the output of `cmake --version` on your system?

I get the same error.
cmake version 3.30.1 on Windows 11

**Reply by sunxuanyuhello** [reply]
> Thanks! This is a relatively simple issue to fix on our end, but it's kinda surprising that you hit this, it's happening because we're finding a cmake binary, but it doesn't have any of the expected version outputs. Out of curiosity, what's the output of `cmake --version` on your system?

I get the same error.
cmake version 3.31.4

**Comment by bad-friday** [comment]
There's no cmake its only a link to make.
[/home/swe]make --version         
GNU Make 4.3
Built for arm-unknown-linux-gnueabihf
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**Reply by eli-schwartz** [reply]
That sounds like a really bad idea tbh.

**Comment by bad-friday** [comment]
I'm sorry

**Reply by dcbaker** [reply]
no worries, this is a legitimate bug on our part, but yeah, it's better to not have make symlinked to cmake, they're very different programs

**Reply by eli-schwartz** [reply]
Right, the only consequences of doing it are that things such as meson may mishandle it -- and nothing will actually benefit from such a symlink anyway.

We can fix meson but we can't fix other software.

**Comment by bad-friday** [comment]
I didn't know that cmake is a seperate package beisde of the package build-essential
